### PR TITLE
Add complex inverse function

### LINKED ISSUE #99: RFC: inverse of a complex number
## Checklist

> Please ensure the following tasks are completed before filing an issue.

* [x] Read and understood the [Code of Conduct][code-of-conduct].
* [x] Searched for existing issues and pull requests.
* [x] If this is a general question, searched the [FAQ][faq] for an existing answer.
* [x] If this is a feature request, the issue name begins with `RFC: `.

## Description

> Description of the issue (or feature request).

Add support for computing the inverse (reciprocal) of a complex number. See https://en.wikipedia.org/wiki/Complex_number#Reciprocal.

## Related Issues

> Does this issue (or feature request) have any related issues?

No.

## Questions

> Any questions for reviewers?

No.

## Other

> Any other information relevant to this issue (or feature request)? This may include screenshots, references, stack traces, sample output, and/or implementation notes.

No.

[code-of-conduct]: https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md
[faq]: https://github.com/stdlib-js/stdlib/blob/develop/FAQ.md

### PR DESCRIPTION
**Pull request by rreusser**
Resolves addition of `complex/inv`, as discussed. Resolves #99.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

* [x] Read, understood, and followed the [contributing guidelines][contributing], including the relevant style guides.
* [x] Read and understand the [Code of Conduct][code-of-conduct].
* [x] Read and understood the [licensing terms][license].
* [x] Searched for existing issues and pull requests __before__ submitting this pull request.
* [x] Filed an issue (or an issue already existed) __prior to__ submitting this pull request.
* [x] Rebased onto latest `develop`.
* [x] Submitted against `develop` branch.

## Description

> What is the purpose of this pull request?

This pull request:

* add a complex inverse function using logic distilled from `complex/divide`. That is, it's mostly just complex division code evaluated for `re1 = 1, im1 = 0` in a manner that matches the [julia code](https://github.com/JuliaLang/julia/blob/0df3f7dc0280525349ae23e3ac4fc96621befc50/base/complex.jl#L378).

## Questions

> Any questions for reviewers of this pull request?

- I left the references intact since if you want to dig into this function, your best bet would be to read and simplify the complex division references. Does that seem reasonable?
- "Compute the inverse of a complex number." I'm sure you'll let me know how you feel about that tense/voice.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

---

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md
[code-of-conduct]: https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md
[license]: https://github.com/stdlib-js/stdlib/blob/develop/LICENSE

### CODE REVIEW

**Comment by kgryte** [comment]
Awesome!

File: lib/node_modules/@stdlib/math/base/complex/inv/docs/repl.txt
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/complex/inv/docs/repl.txt]: s/Compute/Computes/

File: lib/node_modules/@stdlib/math/base/complex/inv/benchmark/c/benchmark.c
Comment (rreusser) [inline lib/node_modules/@stdlib/math/base/complex/inv/benchmark/c/benchmark.c:92]: Is this the best/an appropriate way to compute a complex inverse in c?

File: lib/node_modules/@stdlib/math/base/complex/inv/lib/cinv.js
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/complex/inv/lib/cinv.js]: s/Compute/Computes/

File: lib/node_modules/@stdlib/math/base/complex/inv/lib/main.js
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/complex/inv/lib/main.js]: s/Compute/Computes/

File: lib/node_modules/@stdlib/math/base/complex/inv/test/test.js
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/complex/inv/test/test.js]: Order variables according to length.

File: lib/node_modules/@stdlib/math/base/complex/inv/test/test.js
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/complex/inv/test/test.js]: Order variables according to length.

File: lib/node_modules/@stdlib/math/base/complex/inv/test/test.js
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/complex/inv/test/test.js]: Order variables according to length.

File: lib/node_modules/@stdlib/math/base/complex/inv/test/test.js
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/complex/inv/test/test.js]: Order variables according to length.

File: lib/node_modules/@stdlib/math/base/complex/inv/test/test.js
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/complex/inv/test/test.js]: Order variables according to length.

File: lib/node_modules/@stdlib/math/base/complex/inv/test/test.js
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/complex/inv/test/test.js]: Order variables according to length.

File: lib/node_modules/@stdlib/math/base/complex/inv/test/test.js
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/complex/inv/test/test.js]: Order variables according to length.

File: lib/node_modules/@stdlib/math/base/complex/inv/test/test.js
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/complex/inv/test/test.js]: Order variables according to length.

File: lib/node_modules/@stdlib/math/base/complex/inv/test/test.js
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/complex/inv/test/test.js]: Order variables according to length.

**Review by kgryte — CHANGES_REQUESTED** [review]

**Comment by kgryte** [comment]
We only use the imperative mood (e.g., `Compute`) in three places:

1. Top description in the README, underneath the README first-level heading.
2. In the package.json.
3. In the JSDoc comment in index.js where we provide the module definition and examples.

Everywhere else, we use active voice (e.g., `Computes`).

File: lib/node_modules/@stdlib/math/base/complex/inv/benchmark/c/benchmark.c
Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/complex/inv/benchmark/c/benchmark.c:92]: C does not appear to have a dedicated function for computing the inverse (see http://en.cppreference.com/w/c/numeric/complex). Did you check that `z2` is sensible? If not, may need to implement in the benchmark itself.

File: lib/node_modules/@stdlib/math/base/complex/inv/lib/main.js
Comment (rreusser) [reply lib/node_modules/@stdlib/math/base/complex/inv/lib/main.js]: Changed throughout. (where appropriate) :+1:

Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/complex/inv/lib/main.js]: Awesome. Thanks!

File: lib/node_modules/@stdlib/math/base/complex/inv/test/test.js
Comment (rreusser) [reply lib/node_modules/@stdlib/math/base/complex/inv/test/test.js]: Sigh, find replace. 😑 Fixed.

Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/complex/inv/test/test.js]: Awesome. Thanks!

File: lib/node_modules/@stdlib/math/base/complex/inv/benchmark/c/benchmark.c
Comment (rreusser) [reply lib/node_modules/@stdlib/math/base/complex/inv/benchmark/c/benchmark.c:92]: Yes, the result is correct.

Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/complex/inv/benchmark/c/benchmark.c:92]: Then should be fine to leave as is.

File: lib/node_modules/@stdlib/math/base/complex/inv/lib/cinv.js
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/complex/inv/lib/cinv.js]: `out` is not optional here, so should remove the square brackets.

**Review by Planeshifter — COMMENTED** [review]
Thanks Ricky for working on this!

File: lib/node_modules/@stdlib/math/base/complex/inv/lib/cinv.js
Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/complex/inv/lib/cinv.js]: Good catch!

Comment (rreusser) [reply lib/node_modules/@stdlib/math/base/complex/inv/lib/cinv.js]: Which is then also an issue [here](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/complex/divide/lib/cdiv.js#L26). Trivial to amend, but is it best if PRs target only one module at a time?

Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/complex/inv/lib/cinv.js]: I will fix that locally now.

**Review by kgryte — APPROVED** [review]

**Comment by kgryte** [comment]
Will wait for Circle CI to finish before squashing and merging. Thanks for your work on this @rreusser!

**Pull request merged**
